src/migrations/20220427130214-add-sats-amount-prop-to-payments.ts

```diff
@@ -0,0 +1,99 @@
+module.exports = {
+  async up(db) {
+    const collection = db.collection("medici_transactions")
+    try {
+      const resultDebits = await collection.update(
+        { type: "payment", debit: { $gt: 0 } },
+        [{ $set: { satsAmount: { $subtract: ["$debit", "$fee"] } } }],
+        { multi: true },
+      )
+      console.log(
+        { result: resultDebits },
+        "added new props to debit medici_transactions",
+      )
+    } catch (error) {
+      console.log(
+        { result: error },
+        "Couldn't add new props to debit medici_transactions",
+      )
+    }
+
+    try {
+      const allTxns = await collection.aggregate([
+        {
+          $match: {
+            satsAmount: { $exists: true },
+          },
+        },
+        {
+          $group: {
+            _id: "$hash",
+            satsAmount: { $first: "$satsAmount" },
+            fee: { $first: "$fee" },
+            usd: { $first: "$usd" },
+            feeUsd: { $first: "$feeUsd" },
+          },
+        },
+      ])
+      for await (const { _id: hash, satsAmount, fee, usd, feeUsd } of allTxns) {
+        if (!hash) continue
+        const centsAmount = Math.round((usd - feeUsd) * 100)
+        const centsFee = Math.round(feeUsd * 100)
+
+        const resultRest = await collection.update(
+          { hash },
+          [
+            {
+              $set: {
+                satsAmount,
+                satsFee: fee,
+                centsAmount,
+                centsFee,
+                displayAmount: centsAmount,
+                displayFee: centsFee,
+                displayCurrency: "USD",
```

Comment:
we need to get this value from config, this will not work for bitcoin jungle

Reply:
Good catch. I can change this in other places in the code, but not sure how to import into the migration script here.

Also, it looks like bitcoin jungle is still storing usd on the backend ([here](https://github.com/Bitcoin-Jungle/galoy/blob/f3a927b41101fe07a06df552ea606163328b361c/src/services/ledger/index.ts#L610)). If this is correct, then the migration might be ok like this? They'll likely need a separate migration in the future to migrate their past transactions from usd to crc if that's the case.

Reply:
they dont use USD they use CRC but they have to use usd field (they only changed price server to return CRC) but if we want the migration to work with their data then the default value for them should be CRC

Reply:
Done: https://github.com/GaloyMoney/galoy/pull/1269/commits/7ca4849bfc2943903f0b33292a6707915cac9901